zproject/backends.py

```diff
@@ -866,6 +886,45 @@ def sync_custom_profile_fields_from_ldap(
         except SyncUserException as e:
             raise ZulipLDAPException(str(e)) from e
 
+    def sync_groups_from_ldap(self, user_profile: UserProfile, ldap_user: _LDAPUser) -> None:
```

Comment:
With the current design of how the settings get configured, this should return early without doing anything unless `user_profile.realm` is the realm for which the group sync is enabled

Reply:
see fixup 67e1f39c455b409fcafcb2cabf39b1abf0879250